### ISSUE TITLE
Dealing with unresolvable versions

### ISSUE THREAD

**Issue by ajwheeler** [cjdoris/julia-downgrade-compat-action#6]
Say your compat looks like this:
```
[compat]
A = "1.5"
B = "2.2"
```
but package `B`'s compat is
```
[compat]
A = "1.6"
```
Does applying this action yield unresolvable package versions?

I think this may be the issue that I'm having [here](https://github.com/ajwheeler/Korg.jl/actions/runs/7901130558/job/21564085718?pr=262), for `A`=ForwardDiff and `B`=SpecialFunctions.

**Comment by simonmandlik** [comment]
I'm dealing with a similar problem (the run is [here](https://github.com/CTUAvastLab/Mill.jl/actions/runs/7942453973/job/21685768152)).

I have `Statistics = 1` in `Project.toml`. Since this is a really stable package without much versions that ships with julia, I do not care much about individual versions and want to use whichever version is compatible with the current julia version. However, this action forces `Statistics` to be exactly `1.0`, which AFAIK doesn't even exist.

**Comment by cjdoris** [comment]
@ajwheeler yes that's correct, unless there is some older version of B which has a lower compat on A.

**Comment by cjdoris** [comment]
@simonmandlik you should use the `skip` option, see the readme.

**Comment by simonmandlik** [comment]
@cjdoris thank you. I have seen `skip` in the readme, but I was wondering if the action could use the lowest *existing* version compatible with compat. This would require cloning the registry and is not that trivial I guess. So if I have `A = "1"` in the compat, it will be tested strictly with `1.0.0` version and if it doesn't exist it will fail?

**Comment by cjdoris** [comment]
As it says in the readme, if you have `A = "1"` this gets turned into `A = "~1.0.0"` by this action, meaning that any `1.0.*` version could be installed. It would be very unusual for `1.0.*` to not exist.

**Comment by cjdoris** [comment]
BTW this action has moved to https://github.com/julia-actions/julia-downgrade-compat, I'm about to archive this repo.
